SimMuon/GEMDigitizer/src/ME0SimpleModel.cc

```diff
@@ -0,0 +1,286 @@
+    averageNoiseRatePerRoll *= instLumi_*rateFact_*1.0/referenceInstLumi_;  
+  }
+
+  //simulate intrinsic noise
+  if(simulateIntrinsicNoise_)
+  {
+    const double aveIntrinsicNoisePerStrip(averageNoiseRate_ * nBxing * bxwidth_ * trStripArea * 1.0e-9);
+    for(int j = 0; j < nstrips; ++j)
+    {
+      int randPoissonQ = CLHEP::RandPoissonQ::shoot(engine, aveIntrinsicNoisePerStrip); 
+    
+      for (int k = 0; k < randPoissonQ; k++ )
+      {
+        const int time_hit(static_cast<int>(CLHEP::RandFlat::shoot(engine, nBxing)) + minBunch_);
+	strips_.emplace(k+1,time_hit);
+      }
+    }
+  }//end simulate intrinsic noise
+
+  //simulate bkg contribution
+  const double averageNoise(averageNoiseRatePerRoll * nBxing * bxwidth_ * trArea * 1.0e-9);
+  int randPoissonQ = CLHEP::RandPoissonQ::shoot(engine, averageNoise);
+
+  for (int i = 0; i < randPoissonQ; ++i)
+  {
+    const int centralStrip(static_cast<int> (CLHEP::RandFlat::shoot(engine, 1, nstrips)));
+    const int time_hit(static_cast<int>(CLHEP::RandFlat::shoot(engine, nBxing)) + minBunch_);
+    if (doNoiseCLS_)
+    {
+      std::vector < std::pair<int, int> > cluster_;
+      cluster_.clear();
+      cluster_.emplace_back(centralStrip, time_hit);
+      int clusterSize((CLHEP::RandFlat::shoot(engine)) <= 0.53 ? 1 : 2);
+      if (clusterSize == 2)
+      {
+        if(CLHEP::RandFlat::shoot(engine) < 0.5)
+        {
+          if (CLHEP::RandFlat::shoot(engine) < averageEfficiency_ && (centralStrip - 1 > 0))
+            cluster_.emplace_back(centralStrip - 1, time_hit);
+        }
+        else
+        {
+          if (CLHEP::RandFlat::shoot(engine) < averageEfficiency_ && (centralStrip + 1 <= nstrips))
+            cluster_.emplace_back(centralStrip + 1, time_hit);
+        }
+      }
+      for (const auto& digi : cluster_)
+      {
+	strips_.emplace(digi);
+      }
+    } //end doNoiseCLS_
+    else
+    {
+      strips_.emplace(centralStrip, time_hit);
+    }
+  }
+  return;
+}
+
+std::vector<std::pair<int, int> > ME0SimpleModel::simulateClustering(const ME0EtaPartition* roll,
+    const PSimHit* simHit, const int bx, CLHEP::HepRandomEngine* engine)
+{
+  const StripTopology& topology = roll->specificTopology();
+  const LocalPoint& hit_position(simHit->localPosition());
+  const int nstrips(roll->nstrips());
+  int centralStrip = 0;
+  if (!(topology.channel(hit_position) + 1 > nstrips))
+    centralStrip = topology.channel(hit_position) + 1;
+  else
+    centralStrip = topology.channel(hit_position);
+  const GlobalPoint& pointSimHit = roll->toGlobal(hit_position);
+  const GlobalPoint& pointDigiHit = roll->toGlobal(roll->centreOfStrip(centralStrip));
+  double deltaX = pointSimHit.x() - pointDigiHit.x();
+
+  // Add central digi to cluster vector
+  std::vector < std::pair<int, int> > cluster_;
+  cluster_.clear();
+  cluster_.emplace_back(centralStrip, bx);
+
+  //simulate cross talk
+  int clusterSize((CLHEP::RandFlat::shoot(engine)) <= 0.53 ? 1 : 2);
```

Comment:
"The probabilities of 53% for a single fired strip and 47% for two adjacent fired strips are chosen according to the first test results obtained in 2013 with additional tuning in order to have the cluster size of GEM RecHit relevant to the experimental measured ones."